### PR TITLE
Supply URLs to suite and task event handlers.

### LINKED ISSUE #2275: Pass suite and task URLs to event handlers
These should be optional command line arg templates, like `%(event)s %(suite)s '%(message)s'`

### PR DESCRIPTION
**Pull request by hjoliver**
Close #2275 

~~Not for review yet - just needs tests.~~  Tests added (or rather, existing tests upgraded).

### CODE REVIEW

File: lib/cylc/task_proxy.py [hunk 1]
```diff
@@ -316,9 +320,6 @@ def __init__(
             self.TABLE_TASK_STATES: [],
         }
 
-        # TODO - should take suite name from config!
-        self.suite_name = os.environ['CYLC_SUITE_NAME']
```
Comment (hjoliver) [inline lib/cylc/task_proxy.py]: Doesn't need to be an instance variable, and this was an insane way of getting the suite name.

File: tests/events/08-task-event-handler-retry.t [hunk 2]
```diff
@@ -23,7 +23,7 @@ OPT_SET=
 if [[ "${TEST_NAME_BASE}" == *-globalcfg ]]; then
     create_test_globalrc "" "
 [task events]
-    handlers=hello-event-handler '%(name)s' '%(event)s'
+    handlers = hello-event-handler %(name)s %(event)s %(suite_url)s %(task_url)s %(message)s %(point)s %(submit_num)s %(id)s
```
Comment (hjoliver) [inline tests/events/08-task-event-handler-retry.t]: Might as well test all possible args.

**Comment by hjoliver** [comment]
Will be very minor conflicts with 2157

**Comment by matthewrmshin** [comment]
> Will be very minor conflicts with #2157.

(The new logic will move to `cylc.task_events_mgr` in #2157.)

File: doc/src/cylc-user-guide/suiterc.tex [hunk 3]
```diff
@@ -1546,6 +1547,8 @@ \subsection{[runtime]}
     \item \%(suite)s: suite name
     \item \%(point)s: cycle point
     \item \%(name)s: task name
+    \item \%(suite_url)s: suite url
+    \item \%(task_url)s: task url
```
Comment (matthewrmshin) [inline doc/src/cylc-user-guide/suiterc.tex]: Should be *URL* (ALL CAPS), to be consistent with line 304 above.

**Comment by matthewrmshin** [comment]
@oliver-sanders please sanity check. I am tempted to just merge this and have the docs fixed when I resolve the conflicts in #2157.

**Comment by hjoliver** [comment]
Changed 'url' to 'URL' in the user guide.

**Comment by oliver-sanders** [comment]
Profiling results of the "diamond" experiment show a good improvement to scaling.

memory
cpu-time

**Pull request merged**